From: Tejun Heo <tj@kernel.org>
Subject: [PATCH] cdrom: don't check CDC_PLAY_AUDIO in cdrom_count_tracks()
References: 369558

According to MMC-3 (or any later versions) READ TOCs are mandatory
commands and have nothing to do with CDC_PLAY_AUDIO.  I have no idea
why the check was put there in the first place but it now only breaks
automatic actions on certain drives.

Note that this test was only effective when ide-cdrom was being used
as sr didn't mask CDC_PLAY_AUDIO according to the capabilities.

Signed-off-by: Tejun Heo <tj@kernel.org>
Signed-off-by: Tejun Heo <teheo@suse.de>
---
 drivers/cdrom/cdrom.c |    4 ----
 1 file changed, 4 deletions(-)

Index: linux-2.6.25-SL110_BRANCH/drivers/cdrom/cdrom.c
===================================================================
--- linux-2.6.25-SL110_BRANCH.orig/drivers/cdrom/cdrom.c
+++ linux-2.6.25-SL110_BRANCH/drivers/cdrom/cdrom.c
@@ -1457,10 +1457,6 @@ static void cdrom_count_tracks(struct cd
 	tracks->xa=0;
 	tracks->error=0;
 	cdinfo(CD_COUNT_TRACKS, "entering cdrom_count_tracks\n"); 
-        if (!CDROM_CAN(CDC_PLAY_AUDIO)) { 
-                tracks->error=CDS_NO_INFO;
-                return;
-        }        
 	/* Grab the TOC header so we can see how many tracks there are */
 	if ((ret = cdi->ops->audio_ioctl(cdi, CDROMREADTOCHDR, &header))) {
 		if (ret == -ENOMEDIUM)
